api/api_subsystems.go

```diff
@@ -0,0 +1,61 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+type MinerSubsystem int
```

Comment:
Can we name this plural?

Reply:
Why do we want to make this a bitfield instead of just a simple []string?

Reply:
Added `MinerSubsystems []MinerSubsystem`.